### PR TITLE
Profile Update feature added

### PR DESCRIPTION
**Pull request by dotAsh**

### CODE REVIEW

File: LawGuardPro.API/Controllers/UsersController.cs [hunk 1]
```diff
@@ -1,45 +1,54 @@
...
 using MediatR;
 using Microsoft.AspNetCore.Mvc;
 using System.Net;
+using Microsoft.AspNetCore.JsonPatch;
+using LawGuardPro.Application.Features.Settings.Profiles;
 
-namespace LawGuardPro.API.Controllers
+namespace LawGuardPro.API.Controllers;
+
+[Route("api/UsersAuth")]
```
Comment (sftmaruf) [inline LawGuardPro.API/Controllers/UsersController.cs]: use lower-case letters like `api/usersauth`

File: LawGuardPro.API/LawGuardPro.API.csproj [hunk 2]
```diff
@@ -15,6 +15,7 @@
   <ItemGroup>
     <PackageReference Include="MediatR" Version="12.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="8.0.5" />
+    <PackageReference Include="Microsoft.AspNetCore.JsonPatch" Version="8.0.5" />
```
Comment (sftmaruf) [inline LawGuardPro.API/LawGuardPro.API.csproj]: Remove this if not needed

File: LawGuardPro.Application/Common/Result.cs [hunk 3]
```diff
@@ -31,14 +27,14 @@ protected Result(int statusCode, List<Error> errors)
         }
         return false;
     }
+
     public static Result Success(int statusCode) { 
         return new Result(statusCode, new List<Error>());
     }
+
     public static Result Failure(int statusCode,  List<Error> errors) {
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs]: Remove unnecessary space between two parameters

File: LawGuardPro.Application/Common/Result.cs [hunk 4]
```diff
@@ -1,10 +1,6 @@
-﻿using Microsoft.AspNetCore.Authentication;
+﻿using System.Linq;
 using Microsoft.AspNetCore.Http;
-using System;
 using System.Collections.Generic;
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs:3]: Remove unnecessary usings

File: LawGuardPro.Application/Common/Result.cs [hunk 5]
```diff
@@ -1,10 +1,6 @@
-﻿using Microsoft.AspNetCore.Authentication;
+﻿using System.Linq;
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs:1]: Remove unnecessary usings

File: LawGuardPro.Application/DTO/UserUpdateDTO.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LawGuardPro.Application.DTO;
```
Comment (sftmaruf) [inline LawGuardPro.Application/DTO/UserUpdateDTO.cs]: Always put a space between class declaration and namespace

File: LawGuardPro.Application/DTO/UserUpdateDTO.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
Comment (sftmaruf) [inline LawGuardPro.Application/DTO/UserUpdateDTO.cs]: Remove unnecessary usings

File: LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs [hunk 8]
```diff
@@ -14,5 +14,7 @@ public interface IIdentityService
         Task<Result<LoginResponseDTO>> Login(LoginRequestDTO loginRequestDTO);
```
Comment (sftmaruf) [inline LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs]: Always put space between two method declaration

File: LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs [hunk 9]
```diff
@@ -14,5 +14,7 @@ public interface IIdentityService
...
 
         Task<Result<UserDTO>> Register(RegistrationRequestDTO registration);
+
+        Task<Result<UserDTO>> UpdateUserInfo(UserUpdateDTO userUpdateDto);
```
Comment (sftmaruf) [inline LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs]: `async` method name should contain postfix **Async**

File: LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
+﻿using LawGuardPro.Application.DTO;
+using MediatR;
+using System.Net;
+using System.Reflection;
+using AutoMapper;
+using LawGuardPro.Application.Common;
+using LawGuardPro.Application.Features.Identity.Interfaces;
+
+namespace LawGuardPro.Application.Features.Settings.Profiles;
```
Comment (sftmaruf) [inline LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs:7]: Always put a space between class declaration and namespace

File: LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+public class ProfileEditCommand : IRequest<Result<UserDTO>>
+{
+    public string FirstName { get; set; }
+    public string LastName { get; set; }
+    public string PhoneNumber { get; set; }
+    public string Email { get; set; }
+
+}
+
```
Comment (sftmaruf) [inline LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs:16]: Remove unnecessary spaces

File: LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    public ProfileEditCommandHandler(IMapper mapper, IIdentityService identityService)
+    {
+        _mapper = mapper;
+        _identityService = identityService;
+    }
+
+    public async Task<Result<UserDTO>> Handle(ProfileEditCommand request, CancellationToken cancellationToken)
+    {
+        return await _identityService.UpdateUserInfo(_mapper.Map<UserUpdateDTO>(request));
+
```
Comment (sftmaruf) [inline LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs]: Remove unnecessary space

File: LawGuardPro.Infrastructure/Identity/IdentityService.cs [hunk 13]
```diff
@@ -117,4 +117,34 @@ public async Task<Result<LoginResponseDTO>> Login(LoginRequestDTO loginRequestDT
         return Result<LoginResponseDTO>.Success(loginResponseDTO);
     }
 
+    public async Task<Result<UserDTO>> UpdateUserInfo(UserUpdateDTO userUpdateDto)
+    {
+        if (userUpdateDto == null)
+        {
+            return Result<UserDTO>.Failure(new List<Error> { new Error() { Message = "Invalid user data.", Code = "" } });
+
```
Comment (sftmaruf) [inline LawGuardPro.Infrastructure/Identity/IdentityService.cs]: Remove unnecessary space

File: LawGuardPro.Infrastructure/Identity/IdentityService.cs [hunk 14]
```diff
@@ -117,4 +117,34 @@ public async Task<Result<LoginResponseDTO>> Login(LoginRequestDTO loginRequestDT
...
+            return Result<UserDTO>.Failure(new List<Error> { new Error() { Message = "Invalid user data.", Code = "" } });
+           
+        }
+
+        var user = await _userManager.FindByEmailAsync(userUpdateDto.Email);
+
+        if (user == null)
+        {
+            var errorMessage = $"User with email '{userUpdateDto.Email}' not found.";
+            return Result<UserDTO>.Failure(new List<Error> { new Error() { Message = errorMessage, Code = "UserNotFound" } });
```
Comment (sftmaruf) [inline LawGuardPro.Infrastructure/Identity/IdentityService.cs]: remove the error message variable and put it into inline

File: LawGuardPro.Infrastructure/Identity/IdentityService.cs [hunk 15]
```diff
@@ -117,4 +117,34 @@ public async Task<Result<LoginResponseDTO>> Login(LoginRequestDTO loginRequestDT
...
+        var result = await _userManager.UpdateAsync(user);
+        if (!result.Succeeded) {
+            var errors = result.Errors.Select(error => new Error { Message = error.Description, Code = error.Code }).ToList();
+            return Result<UserDTO>.Failure(errors);
+        }
+
+        var userDto = _mapper.Map<UserDTO>(user);
+        return Result<UserDTO>.Success(userDto);
+    }
+
```
Comment (sftmaruf) [inline LawGuardPro.Infrastructure/Identity/IdentityService.cs]: Remove unnecessary space

**Review by sftmaruf — CHANGES_REQUESTED** [review]

File: LawGuardPro.API/Controllers/UsersController.cs (hunk 1 above)
Comment (dotAsh) [reply LawGuardPro.API/Controllers/UsersController.cs]: Updated route to use lowercase letters (api/usersauth)

File: LawGuardPro.API/LawGuardPro.API.csproj (hunk 2 above)
Comment (dotAsh) [reply LawGuardPro.API/LawGuardPro.API.csproj]: Removed unnecessary package reference from csproj

File: LawGuardPro.Application/Common/Result.cs (hunk 3 above)
Comment (dotAsh) [reply LawGuardPro.Application/Common/Result.cs]: Done.

File: LawGuardPro.Application/Common/Result.cs (hunk 4 above)
Comment (dotAsh) [reply LawGuardPro.Application/Common/Result.cs:3]: Removed unnecessary using statements

File: LawGuardPro.Application/Common/Result.cs (hunk 5 above)
Comment (dotAsh) [reply LawGuardPro.Application/Common/Result.cs:1]: Removed unnecessary using statements

File: LawGuardPro.Application/DTO/UserUpdateDTO.cs (hunk 6 above)
Comment (dotAsh) [reply LawGuardPro.Application/DTO/UserUpdateDTO.cs]: Done.

File: LawGuardPro.Application/DTO/UserUpdateDTO.cs (hunk 7 above)
Comment (dotAsh) [reply LawGuardPro.Application/DTO/UserUpdateDTO.cs]: Done.

File: LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs (hunk 9 above)
Comment (dotAsh) [reply LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs]: Renamed async methods to include Async suffix

File: LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs (hunk 10 above)
Comment (dotAsh) [reply LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs:7]: Done.

File: LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs (hunk 11 above)
Comment (dotAsh) [reply LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs:16]: Removed unnecessary spaces

File: LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs (hunk 12 above)
Comment (dotAsh) [reply LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs]: Done.

File: LawGuardPro.Infrastructure/Identity/IdentityService.cs (hunk 13 above)
Comment (dotAsh) [reply LawGuardPro.Infrastructure/Identity/IdentityService.cs]: unnecessary space removed.

File: LawGuardPro.Infrastructure/Identity/IdentityService.cs (hunk 15 above)
Comment (dotAsh) [reply LawGuardPro.Infrastructure/Identity/IdentityService.cs]: Inlined error message in IdentityService

File: LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs (hunk 8 above)
Comment (dotAsh) [reply LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs]: Done.

File: LawGuardPro.API/Controllers/UsersController.cs [hunk 16]
```diff
@@ -4,16 +4,14 @@
 using MediatR;
 using Microsoft.AspNetCore.Mvc;
 using System.Net;
```
Comment (sftmaruf) [inline LawGuardPro.API/Controllers/UsersController.cs]: Still has unused usings.

File: LawGuardPro.API/Controllers/UsersController.cs [hunk 17]
```diff
@@ -4,16 +4,14 @@
...
 
 namespace LawGuardPro.API.Controllers;
 
-[Route("api/UsersAuth")]
+[Route("api/usersauth")]
 [ApiController]
 public class UsersController : ControllerBase
 {
-
     private readonly ISender _sender;
```
Comment (sftmaruf) [inline LawGuardPro.API/Controllers/UsersController.cs:12]: Always put a space between constructor and class variable.

File: LawGuardPro.API/Controllers/UsersController.cs [hunk 18]
```diff
@@ -40,15 +38,11 @@ public async Task<IActionResult> Login([FromBody] UserLoginCommand model)
     [HttpPatch("UpdateUserInfo")]
     public async Task<IActionResult> UpdateUserInfo(ProfileEditCommand model)
```
Comment (sftmaruf) [inline LawGuardPro.API/Controllers/UsersController.cs:38]: Always put a space between two methods

File: LawGuardPro.Application/Common/Result.cs [hunk 19]
```diff
@@ -14,25 +14,21 @@ public class Result : IResult
 {
     public int StatusCode { get; set; }
     public List<Error> Errors { get; set; }
-
     protected Result(int statusCode, List<Error> errors)
     {
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs]: Always put a space between constructor and class variable.

File: LawGuardPro.Application/Common/Result.cs [hunk 20]
```diff
@@ -14,25 +14,21 @@ public class Result : IResult
...
         StatusCode = statusCode;
         Errors = errors;
     }
-
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs]: Always put a space between two constructors

File: LawGuardPro.Application/Common/Result.cs [hunk 21]
```diff
@@ -14,25 +14,21 @@ public class Result : IResult
...
     public bool IsSuccess() { 
         if(StatusCode >= 200 || StatusCode <= 300){
             return true;
         }
         return false;
     }
-
     public static Result Success(int statusCode) {
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs:31]: Always put a space between two methods

File: LawGuardPro.Application/Common/Result.cs [hunk 22]
```diff
@@ -49,13 +45,11 @@ public class Result<T> : IResult<T>
     public T? Data { get; }
     public int StatusCode { get; }
     public List<Error> Errors { get; }
-
     protected Result(int statusCode, T? data, List<Error> errors) {
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs:53]: Always put a space between constructor and class variable.

File: LawGuardPro.Application/Common/Result.cs [hunk 23]
```diff
@@ -64,12 +58,10 @@ public bool IsSuccess()
         }
         return false;
     }
-
     public static Result<T> Success(T data) {
```
Comment (sftmaruf) [inline LawGuardPro.Application/Common/Result.cs:68]: Always put a space between two methods

File: LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs [hunk 24]
```diff
@@ -1,20 +1,16 @@
 ﻿using LawGuardPro.Application.Common;
 using LawGuardPro.Application.DTO;
-using System;
 using System.Collections.Generic;
```
Comment (sftmaruf) [inline LawGuardPro.Application/Features/Identity/Interfaces/IIdentityService.cs:3]: Still has unused usings

File: LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs [hunk 25]
```diff
@@ -28,11 +24,9 @@ public ProfileEditCommandHandler(IMapper mapper, IIdentityService identityServic
         _mapper = mapper;
         _identityService = identityService;
     }
-
     public async Task<Result<UserDTO>> Handle(ProfileEditCommand request, CancellationToken cancellationToken)
```
Comment (sftmaruf) [inline LawGuardPro.Application/Features/Settings/Profiles/ProfileEditCommand.cs:28]: Always put a space between two methods

File: LawGuardPro.Infrastructure/Identity/IdentityService.cs [hunk 26]
```diff
@@ -75,18 +68,15 @@ public async Task<Result<UserDTO>> Register(RegistrationRequestDTO registrationR
     }
```
Comment (sftmaruf) [inline LawGuardPro.Infrastructure/Identity/IdentityService.cs]: Remove extra spaces

File: LawGuardPro.Infrastructure/Identity/IdentityService.cs [hunk 27]
```diff
@@ -1,21 +1,15 @@
...
-using System.IdentityModel.Tokens.Jwt;
-using System.Text;
-using Microsoft.AspNetCore.SignalR;
 using Microsoft.IdentityModel.Tokens;
-using LawGuardPro.Application.Common;
-using System.Net;
-using Microsoft.AspNetCore.Http;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
 using LawGuardPro.Application.Features.Identity.Interfaces;
```
Comment (sftmaruf) [inline LawGuardPro.Infrastructure/Identity/IdentityService.cs]: Remove extra space.

**Review by sftmaruf — CHANGES_REQUESTED** [review]

**Review by sftmaruf — APPROVED** [review]

**Pull request merged**